### PR TITLE
[OP#41127] filter work packages by storage and sort by status

### PR DESCRIPTION
**Pull request by individual-it**
when searching by query (to link a wp to file):
- don't filter by a magic status id, but by the meta-status "open" 
- additionally filter out packages that cannot be linked to the current storage

CC @Kharonus

### CODE REVIEW

**Review by kiranparajuli589 — APPROVED** [review]
lgtm :+1:

File: lib/Service/OpenProjectAPIService.php [hunk 1]
```diff
@@ -236,15 +236,26 @@ public function getNotifications(
...
+		string $userId,
+		string $query = null,
+		int $fileId = null,
+		string $storageUrl = null,
+		int $offset = 0,
+		int $limit = 5
 	): array {
+		$linkableStorageFilter = [
+			'linkable_to_storage_url' =>
+				['operator' => '=', 'values' => [$storageUrl]]
```
Comment (individual-it) [inline lib/Service/OpenProjectAPIService.php:256]: ToDo: urlencode

**Comment by individual-it** [comment]
@Kharonus encoding the URL as requested by you. Could you please review the filters

**Comment by Kharonus** [comment]
> @Kharonus encoding the URL as requested by you. Could you please review the filters

The used filters look correct from syntax and semantic. Just one question remains: Should we include now the closed tickets in the search result? And add a `sortBy` to the request?

**Comment by individual-it** [comment]
@Kharonus @kiranparajuli589 I've changed it so, that its now does not filter by status at all, but only by the storage and shows results of closed status at the end.
Please review again

**Comment by individual-it** [comment]
don't undestand why CI is failing, works locally also with newest nextcloud, need to investigate

**Comment by individual-it** [comment]
CI fails because of this change in NC master https://github.com/nextcloud/server/commit/6be7aa112f6174c1406cfef5449836124a7f6f50#diff-cd64d65cb5d9cb2c60c6a050ac0739f4b94390011ef66ee81523885f60a4c64bL63
PR: https://github.com/nextcloud/server/pull/31609

File: tests/lib/Service/OpenProjectAPIServiceTest.php [hunk 2]
```diff
@@ -150,16 +151,29 @@ private function getOpenProjectAPIService(
...
-			$this->createMock(\OC\Http\Client\LocalAddressChecker::class)
-		);
+
+		//changed from nextcloud 24
+		if (version_compare(OC_Util::getVersionString(), '24') >= 0) {
+			// @phpstan-ignore-next-line
+			$ocClient = new Client(
+				$this->createMock(IConfig::class),                             // @phpstan-ignore-line
+				$certificateManager,                                                          // @phpstan-ignore-line
+				$client,                                                                      // @phpstan-ignore-line
```
Comment (kiranparajuli589) [inline tests/lib/Service/OpenProjectAPIServiceTest.php:163]: indentations are a little too more here.

File: tests/lib/Service/OpenProjectAPIServiceTest.php [hunk 3]
```diff
@@ -150,16 +151,29 @@ private function getOpenProjectAPIService(
...
+				$this->createMock(\OC\Http\Client\LocalAddressChecker::class)  // @phpstan-ignore-line
+			);
+		} else {
+			// @phpstan-ignore-next-line
+			$ocClient = new Client(
+				$this->createMock(IConfig::class),                             // @phpstan-ignore-line
+				$this->createMock(ILogger::class),                             // @phpstan-ignore-line
+				$certificateManager,                                                          // @phpstan-ignore-line
+				$client,                                                                      // @phpstan-ignore-line
```
Comment (kiranparajuli589) [inline tests/lib/Service/OpenProjectAPIServiceTest.php:172]: indentations are a little too more here

**Review by kiranparajuli589 — COMMENTED** [review]
very minor comments

File: tests/lib/Service/OpenProjectAPIServiceTest.php (hunk 2 above)
Comment (individual-it) [reply tests/lib/Service/OpenProjectAPIServiceTest.php:163]: moved the fix to https://github.com/nextcloud/integration_openproject/pull/95 and fixed the indent there, please review there

**Pull request merged**